### PR TITLE
Fixing peer review issues

### PR DESCRIPTION
**Pull request by asmeurer**
See https://github.com/sympy/sympy-paper/issues/204. Feel free to push to this branch. Please make a note of which items are fixed in your commit messages so we can check them off. 

Here is the PeerJ guide to the rebuttal letter https://peerj.com/benefits/academic-rebuttal-letters/

### CODE REVIEW

**Comment by asmeurer** [comment]
The rebuttal letter needs a cover letter. We should also copy and italicize all the reviewer comments into the letter.

File: paper.bib [hunk 1]
```diff
@@ -3577,7 +3577,7 @@ @incollection{dsl-little-languages
              Vol.\ III: Little Languages and Tools},
   chapter=3,
   pages={39-60},
-  editors={Peter H.\ Salas},
+  editor ={Peter H.\ Salas},
```
Comment (asmeurer) [inline paper.bib:3588]: Oh derp.

File: rebuttal.tex [hunk 2]
```diff
@@ -0,0 +1,730 @@
...
+simplification (see last input line --- \texttt{exp(100)} cancel).
+Second example is good, but our looks to be more simple.
+
+Applying \texttt{evalf()} has nothing related to \emph{symbolic} computation,
+that method is for numerical evaluation, that should be clear from
+the sentence below ``The evalf method converts a constant symbolic expression
+to a Float\dots''.  Numerical evaluation of the whole expression should
+always solve cancellation problem, as it was stated in the sentence
+``Applying the evalf method to the whole expression solves this problem.''
+\end{solution}
```
Comment (asmeurer) [inline rebuttal.tex:736]: @fredrik-johansson can you comment on whether or not it is true that applying `evalf()` to the whole expression should _always_ avoid cancellation issues?

Comment (asmeurer) [inline rebuttal.tex:736]: If so, can you suggest how to improve the wording in the article, and if not, do you have a counterexample?

**Comment by asmeurer** [comment]
@fredrik-johansson it would be helpful if you could review the changes to numerics.tex.

**Comment by certik** [comment]
Working on the speed issue.

**Comment by certik** [comment]
@asmeurer please review my wording in 4507c1591ac60a41bdb6b69571428aed26df0fdd.

**Comment by certik** [comment]
@asmeurer I am done with the "speed" point. Please review my changes.

**Comment by asmeurer** [comment]
Looks good. I added a small change to make it clear that SymEngine isn't used in 1.0 (which is the version used for the paper).

**Comment by certik** [comment]
The acknowledgements were removed in 5ef75ae24816e907f2fe47927b3d7684dbf75b6e, which is not the right fix, so I reverted it in b0d4c326ab43ae1fcada6724eafa7a3bd0cde92c. Then I pushed in a proper fix, as the reviewer asked in b0d4c32...8441004 (this includes two unrelated @asmeurer's commits, but I don't know how to tell github not to include them).

@rpmuller please let me know if this change is ok with you. The reviewer asked:

> Lines 670--678 contain explanations what some of the authors' institutions are. It is customary to use acknowledgements to thank people and institutions, while institutions' details should be provided in the authors' footnotes in the documents' head.

So I did what he asked.

**Comment by asmeurer** [comment]
@certik see https://github.com/sympy/sympy-paper/issues/205 regarding the acknowledgements.

**Comment by certik** [comment]
@asmeurer looks like we still have to modify the last acknowledgement, but my change seems to be consistent with both the reviewer and #205. Let me know if you disagree.

**Comment by asmeurer** [comment]
My understanding is that the acknowledgements section should be removed entirely.

**Comment by rpmuller** [comment]
How can you have a paper without acknowledging the funding sources? You
certainly can't have LANL or SNL authors on it, if you choose to do so.

**Comment by asmeurer** [comment]
@rpmuller see https://peerj.com/about/author-instructions/cs#standard-sections. The funding information _is_ included, but we need to include it in the metadata on the PeerJ site when we (that is, I) submit, not in the article itself. It is then included in the final article by PeerJ. See for example, [this article](https://peerj.com/articles/cs-80/#additional-information).

**Comment by rpmuller** [comment]
Got it. In that case, I'm fine with the changes. Sorry for the confusion.

**Comment by certik** [comment]
@asmeurer, @rpmuller thanks for the clarifying it. I was actually confused too. Looks like that all is good, and it will be acknowledged just fine, I didn't know that PeerJ has a special section for this.

**Comment by certik** [comment]
Regarding CASes, make sure you uncheck the box, since a reviewer asked to fix this.

**Comment by asmeurer** [comment]
I did. I will likely end up sending an email to the editor about this, to see if the journal has a style guide that we should follow.

**Comment by asmeurer** [comment]
Response from the editor on CASs vs. CAS's:

> This is one of the style points that we don't have a set policy on. We just ask that you be consistent throughout the article with whichever style you choose. I'm ccing your AE on this so he is in the loop on this particular point. 

So let's just use CAS's, as that was requested by the reviewer.

**Comment by certik** [comment]
Didn't the reviewer request CASes?

File: rebuttal.tex [hunk 3]
```diff
@@ -357,6 +357,21 @@ \section{Reviewer 3}
 \item The current conclusion works fine as the finishing section of the paper.
 \end{enumerate}
 \begin{solution}
+  FIXME.
+  \begin{enumerate}
+    \item I believe we shouldn't move ``Other Projects'' section to the
+      main paper content.  It either should stay in the supplement or we
+      should do a moderate cleanup for this section.  E.g. list only few
+      projects here and put other in the supplement.  Or/And show here
+      only project names.
```
Comment (asmeurer) [inline rebuttal.tex]: I agree with this, but I think we'll need a stronger argument if we are going to leave it out.

**Comment by asmeurer** [comment]
Does anyone know if it's possible to `\ref` supplement section numbers in the paper?

**Comment by asmeurer** [comment]
I have added some rebuttal to the rewrite point. I would appreciate some feedback on it, though, as I'm sure it could be improved.

**Comment by asmeurer** [comment]
This is ready for people to start reviewing and signing off on.

File: numerics.tex [hunk 4]
```diff
@@ -68,19 +79,30 @@ \subsection{The mpmath library}
 0.100000000000000000000192874985
 \end{verbatim}
 
-Like SymPy, mpmath is a pure Python library.
-Internally, mpmath represents a floating-point number
-${(-1)}^s x \cdot 2^y$ by a tuple $(s, x, y, b)$ where
+Like SymPy, mpmath is a pure Python library.  A design decision of SymPy is to
+keep it and its required dependencies pure Python. This is a primary advantage
+ of mpmath over other multiple precision libraries such as GNU MPFR~\cite{Fousse:2007:MMB:1236463.1236468},
+which is much more feature-rich and fast.  Like SymPy, mpmath is also BSD
```
Comment (fredrik-johansson) [inline numerics.tex]: I don't think MPFR is more feature-rich!

Comment (asmeurer) [reply numerics.tex]: @fredrik-johansson can you push a change to this branch for how you would rather word it (and note that this is in response to a comment by the reviewers, see the rebuttal document).

**Comment by asmeurer** [comment]
@scopatz does the letterhead on the cover letter look good to you?

**Comment by scopatz** [comment]
> @scopatz does the letterhead on the cover letter look good to you?

+1 from me!

File: rebuttal.tex [hunk 5]
```diff
@@ -0,0 +1,948 @@
...
+or two different ways to compute the 100th Fibonacci number:
+
+\begin{verbatim}
+   >>> phi = (1+sqrt(5))/2
+   >>> psi = (1-sqrt(5))/2
+   >>> ((phi**100-psi**100)/sqrt(5)).evalf()-fibonacci(100)
+   65536.0000000000
+   >>> ((phi**100-psi**100)/sqrt(5) - fibonacci(100)).evalf()
+   0.e-104
+\end{verbatim}
```
Comment (asmeurer) [inline rebuttal.tex:724]: @fredrik-johansson how do you feel about the example currently in the paper (`cos(exp(-100)) - 1`) vs. this one? The paper states that this is a "notorious case". Is this expression a classic example, which can be cited from somewhere?

**Comment by asmeurer** [comment]
Another thing we should do is check all the URLs in the documents to check that they are valid. Does anyone know of a tool that will check all URLs to make sure they don't 404?

File: paper.bib [hunk 6]
```diff
@@ -3590,3 +3598,149 @@ @misc{mpmath
...
+  note         = {\url{http://www.sympygamma.com/}},
+  Year         = {2016},
+}
+
+@Misc{scipy,
+  author =    {Eric Jones and Travis Oliphant and Pearu Peterson and others},
+  title =     {{SciPy}: Open source scientific tools for {Python}},
+  year =      {2001--},
+  url = "http://www.scipy.org/",
+  note = {[Online; accessed 2016-09-28]}
```
Comment (asmeurer) [inline paper.bib:3723]: Should we get rid of this (it shows up in the citation)?

**Comment by asmeurer** [comment]
I've read through the paper and the rebuttal, and I'm +1 to them (although I still can't decide if we should just use "CASs", c.f. https://github.com/sympy/sympy-paper/commit/58453435aa7c377952461391e8c0776650bac625#commitcomment-19230766). Only question I have: should we use cross-document references to reference specific section numbers when we mention the supplement?

**Comment by asmeurer** [comment]
Why is the Brent and Zimmerman reference in the supplement (in the series section) rendering as `[Brent and Zimmerman]` instead of a number?

**Comment by asmeurer** [comment]
I am also +1 to the supplement.

**Comment by asmeurer** [comment]
Thank you to everyone who helped with the reviews. I am going to resubmit the paper now. Please let me know right away if there are any objections.

**Pull request merged**